src/Catel.MVVM/Catel.MVVM.Shared/MVVM/ViewModels/ViewModelBase.cs

```diff
@@ -201,10 +203,13 @@ protected ViewModelBase()
         /// <param name="skipViewModelAttributesInitialization">
         /// if set to <c>true</c>, the initialization will be skipped and must be done manually via <see cref="InitializeViewModelAttributes"/>.
         /// </param>
+        /// <param name="reuseIdentifier">
+        /// if set to <c>true</c>, the identifier will be reused from the relased identifier pool.
+        /// </param>
         /// <exception cref="ModelNotRegisteredException">A mapped model is not registered.</exception>
         /// <exception cref="PropertyNotFoundInModelException">A mapped model property is not found.</exception>
-        protected ViewModelBase(bool supportIEditableObject, bool ignoreMultipleModelsWarning = false, bool skipViewModelAttributesInitialization = false)
-            : this(null, supportIEditableObject, ignoreMultipleModelsWarning, skipViewModelAttributesInitialization)
+        protected ViewModelBase(bool supportIEditableObject, bool ignoreMultipleModelsWarning = false, bool skipViewModelAttributesInitialization = false, bool reuseIdentifier = false)
```

Comment:
I think this is an actual breaking change. We'll need 2 overloads (at least to make sure that already compiled code keeps working with the old ctors).

Reply:
This was my idea.
@alexfdezsauco proposed to remove GetUniqueIdentifier(bool reuse) overload. If someone like me need reuse Ids, he can create "ReuseObjectIdGenerator" and register it before VM initialization.

Reply:
Fixed